rust/src/data_catalog/unity/models.rs

```diff
@@ -0,0 +1,367 @@
+#[derive(Deserialize)]
+#[serde(untagged)]
+pub enum GetSchemaResponse {
+    /// Successful response
+    Success(Box<Schema>),
+    /// Error response
+    Error(ErrorResponse),
+}
+
+/// List table summaries response
+#[derive(Deserialize)]
+#[serde(untagged)]
+pub enum ListTableSummariesResponse {
+    /// Successful response
+    Success {
+        /// Basic table infos
+        tables: Vec<TableSummary>,
+        /// Continuation token
+        next_page_token: Option<String>,
+    },
+    /// Error response
+    Error(ErrorResponse),
+}
+
+/// A schema within a catalog
+#[derive(Deserialize)]
+pub struct Schema {
+    /// Username of schema creator.
+    pub created_by: String,
+
+    /// Name of schema, relative to parent catalog.
+    pub name: String,
+
+    /// Username of user who last modified schema.
+    pub updated_by: String,
+
+    /// Full name of schema, in form of catalog_name.schema_name.
+    pub full_name: String,
+
+    /// The type of the parent catalog.
+    pub catalog_type: String,
+
+    /// Name of parent catalog.
+    pub catalog_name: String,
+
+    /// Storage root URL for managed tables within schema.
+    pub storage_root: String,
+
+    /// Storage location for managed tables within schema.
+    pub storage_location: String,
+
+    /// A map of key-value properties attached to the securable.
+    pub properties: HashMap<String, String>,
+
+    /// User-provided free-form text description.
+    pub comment: String,
+
+    /// Time at which this schema was created, in epoch milliseconds.
+    pub created_at: i64,
+
+    /// Username of current owner of schema.
+    pub owner: String,
+
+    /// Time at which this schema was created, in epoch milliseconds.
+    pub updated_at: i64,
+
+    /// Unique identifier of parent metastore.
+    pub metastore_id: String,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[allow(missing_docs)]
+/// Possible data source formats for unity tables
+pub enum DataSourceFormat {
+    Delta,
+    Csv,
+    Json,
+    Avro,
+    Parquet,
+    Orc,
+    Text,
+    UnityCatalog,
+    Deltasharing,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+#[allow(missing_docs)]
+/// Possible data source formats for unity tables
+pub enum TableType {
+    Managed,
+    External,
+    View,
+    MaterializedView,
+    StreamingTable,
+}
+
+///
+#[derive(Deserialize)]
+pub struct TableSummary {
+    /// The full name of the table.
+    pub full_name: String,
+    /// Type of table
+    pub table_type: TableType,
+}
+
+/// A table within a schema
+#[derive(Deserialize)]
+pub struct Table {
+    /// Username of table creator.
+    pub created_by: String,
+
+    /// Name of table, relative to parent schema.
+    pub name: String,
+
+    /// Username of user who last modified the table.
+    pub updated_by: String,
+
+    /// List of schemes whose objects can be referenced without qualification.
+    pub sql_path: String,
```

Comment:
This is the culprit. My Unity Catalog returns response without any `sql_path`.
```suggestion
    #[serde(default)]
    pub sql_path: String,
```
I fixed it like this but looking at the documentation, it looks like most of the response fields could be optional (even though it doesn't make much sense in some cases).